nf_core/modules/update.py

```diff
@@ -33,7 +35,21 @@ def __init__(
         self.show_diff = show_diff
         self.save_diff_fn = save_diff_fn
 
+    def _parameter_compatibility_check(self):
+        """Check the compatibilty of the supplied parameters.
+
+        Checks:
+            - Either `--preview` or `--save_diff` can be specified, not both.
+        """
+
+        # Can't this be handled by click with mutually exclusive parameters?
```

Comment:
That would be nice, can you do this with native click functionality?

Reply:
I never really used click before and a quick search for mutually exclusive arguments didn't result in any "native click" solution. Can you give ne a pointer to where that has been already implemented in this repo or where I can find it in the click docs? The best I could find on the official docs was https://click.palletsprojects.com/en/8.1.x/options/#callbacks-for-validation, but that seems tedious and not very streamlined, and this solution https://stackoverflow.com/a/37491504/6018688 which I like better.

Reply:
Ok, from the way you wrote the comment I thought you meant that you knew of a better solution.

No, I've not seen this in native Click myself and to be honest I think I'd prefer to keep the logic here rather than start messing with custom `cls` elements in Click. This is only a couple of lines of code, the Click stuff would be pretty complex by comparison. Also it has repercussions in how we use `rich_click`, also making that syntax messier.